### DISCUSSION TITLE
mpremote, how do you edit files?

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#10904] [Using MicroPython] [answered]
After many attempts to edit a file I "managed" to do it once and ended with vim as the editor.  It has been 15 years since I used it and have tried to change it to nano.

So, into `.bashrc` and place `EDITOR=/usr/bin/nano` then `source ~/.bashrc`

Next `mpremote edit my_file.txt` and I get `mpremote: 'edit' is not a command`

I can do other stuff like `mpremote fs ls`

It must be something pretty basic as the only mention in the docs is ... 

> (defined by the environment variable $EDITOR)

What have I forgotten to do? (Ubuntu 20.04LTS)

**Comment by robert-hh** [comment, marked as answer]
I have the line:
`export EDITOR="/usr/bin/subl -w"`
in my .bashrc file, which works. I'm using sublime text, called with `subl`

**Reply by davefes** [reply]
Tried:
`export EDITOR='usr/bin/nano'`
and 
`export EDITOR='usr/bin/nano -w'`
with the same error.

**Reply by davefes** [reply]
```py
dave@davef:~$ sudo mpremote edit errors.txt
[sudo] password for dave: 
sudo: mpremote: command not found
```
Thought I'd try sudo.

**Reply by robert-hh** [reply]
The -w is a specialty of Sublime Text. Nano should not need that. Using `export EDITOR="/usr/bin/nano"` Nano starts fine. The difference I see: I use double quotes, you use single quotes. But that does not matter. It works as well without quotes.
Did you restart the bash window after the changes?

**Reply by robert-hh** [reply]
What happens if you type:
`EDITOR=/usr/bin/nano mpremote edit  errors.txt`

**Reply by davefes** [reply]
Tried single and double quotes and yes I run `source ~./bashrc`each time.
```py
dave@davef:~$ EDITOR=/usr/bin/nano mpremote edit errors.txt
mpremote: 'edit' is not a command
```

**Reply by robert-hh** [reply]
Which version of mpremote do you use? I'm getting 0.4.0 calling `mpremote --version`. And what is the result of `which nano`?
Interestingly, I have two copies of nano on my system.

**Reply by davefes** [reply]
Ah mpremote 0.3.0  I had today tried installing it and it was already on the machine.  I'll get rid of that version and try again.
```py
dave@davef:~$ which nano
/usr/bin/nano
```

**Reply by davefes** [reply]
Thank you!! I keep my Ubuntu 20.04LTS up to date but looks like one has to do the same with any pip packages.
